third_party/cpgfunctionEP-0.5/src/segments.cpp

```diff
@@ -49,7 +49,7 @@ namespace gt::segments {
             nq = jcc::interpolation::interp1d(drilling_depth,
                                               drilling_depths[0],
                                               ideal_segment_lengths[0]);
-        } else if(heights[0] < height < heights[n1]) {
```

Comment:
In C++, `x < y < z` doesn't do what you think it does.  C++ does not have chained operation.  This will perform one binary comparison, which in this case returns a boolean, then try to compare that boolean to a floating point with the second binary operation.  I'm assuming you just want to do `x < y && y < z`, so that's what I did here.  This nifty syntax works in Python but not C++.